### ISSUE TITLE
Etape 5 - Dénombrements - Ergonomie

### ISSUE THREAD

**Issue by camillemonchicourt** [PnX-SI/gn_mobile_occtax#114]
Je crois que ce sujet est souvent revenu, en tout cas on a souvent des retours de nos agents qui ont du mal avec la compréhension de cette étape car elle a une ergonomie des autres étapes.
Il y a un bouton SAUVEGARDER en haut à droite et non pas en bas à droite comme sur les autres étapes.
Le bouton + n'est pas non plus compris par les agents.

Il faudrait revoir cette étape et son ergonomie pour la rendre plus intuitive et fluide.

**Comment by DonovanMaillard** [comment]
Oui, une homogénéisation et clarification est prévue, et de manière générale une révision un peu globale de l'ergonomie, dans le cadre de l'AAP SINP 2021. On proposera ici les solutions envisagées et retenues.

**Comment by TheoLechemia** [comment]
Autre piste pour les dénombrements : 
Directement sur la page du formulaire des dénombrements, ajouter un bouton " + dénombrement" . Celui-ci aurait pour effet de réduire le bloc de dénombrement courant (composant de type buger ou "expansion panel : https://material.angular.io/components/expansion/overview) et d'ajouter un nouveau formulaire de dénombrement dessous.
Cela permettrait de supprimer l'écran suivant de "bilan" des dénombrement, et aussi d'homogénéisé avec ce qu'on fait en web.

**Comment by DonovanMaillard** [comment]
Pour le moment je suis sur le point de commander une évolution qui va dans le même sens que le reste, à savoir un bouton suivant en bas à droite. Et un bouton "+ Ajout dénombrement" sur la liste des dénombrements. 

A terme, j'aimerais en effet quand on a des listes déroulantes (dataset, observateurs, nomenclatures etc) changer le fonctionnement : actuellement on arrive sur une nouvelle page, on doit cliquer sur la valeur qu'on souhaite, puis faire un retour arrière qui valide le choix. Ca fait 2 clics, ça fait perdre en lisibilité sur les étapes et le fil de la saisie. 
Je voudrais faire évoluer ça pour que chaque liste déroulante apparaisse sur une modale, et que la sélection de la valeur retenue suffise à changer la valeur + fermer la modale. Ca gagne un clic et en lisibilité je pense. Je pense que si on a un fonctionnement comme ca, on comprendra mieux le formulaire dénombrement...

**Comment by TheoLechemia** [comment]
> A terme, j'aimerais en effet quand on a des listes déroulantes (dataset, observateurs, nomenclatures etc) changer le fonctionnement : actuellement on arrive sur une nouvelle page, on doit cliquer sur la valeur qu'on souhaite, puis faire un retour arrière qui valide le choix. Ca fait 2 clics, ça fait perdre en lisibilité sur les étapes et le fil de la saisie.

Je viens d'installé la 2.0.0 et j'ai déjà le comportement que tu souhaite sur les toutes les nomenclatures, JDD. ça ouvre une modale et à la selection, ça la ferme.
Le seul endroit ou j'ai le comportement que tu décrit c'est sur les observateurs, car tu peux sélectionner plusieurs observateurs. Et là, pour moi il faudrait un bouton valider en bas, plutôt que de devoir cliquer sur "retour"

**Comment by DonovanMaillard** [comment]
Ca marche merci, alors je me suis perdu avec les observateurs et mon idée de modale plutôt qu'une nouvelle page à chaque fois...

**Comment by camillemonchicourt** [comment]
Une des évolutions évoquées dans ce ticket a été prise en compte dans la version 2.1.0, en rendant le bouton + des étapes "Dénombrement" et "Taxon" plus explicites, avec un message ajouté sur le bouton : 
![Screenshot_20220504-124757_Occtax](https://user-images.githubusercontent.com/4418840/166672746-0d5b4314-f3eb-447d-a890-3e9cb83f2816.jpg)

![Screenshot_20220504-124842_Occtax](https://user-images.githubusercontent.com/4418840/166672771-280d8086-ff34-4ddb-b62b-d61d6dfccec5.jpg)

Reste à voir les autres sujets évoqués dans ce ticket.

**Comment by TheoLechemia** [comment]
Est ce que le fait que l'étape 5 soit dans une modale qui ne prenne pas tout l'écran a été traité par ce commit ? (ce qui solutionnait le bouton "sauvegarder" en haut à droite, hétérogène par rapport aux autres étape)
Le ticket ne couvrait pas uniquement le fait d'avoir un "+ ajouter un dénombrement"

**Comment by camillemonchicourt** [comment]
En effet, c'est un autre sujet, non traité pour le moment, je réouvre.

**Comment by DonovanMaillard** [comment]
bon de commande en cours de rédaction pour la révision ergo, notamment cette modale :) 

Contrairement à ce qu'on avait imaginé à la base, la prochaine commande portera sur pas mal de points ergo et 2 fonctionnalités simples à implémenter. On ne retouche pas à la synchro dans la prochaine version, le temps de faire un tour de la question et de proposer des choses plus stables et simples à gérer coté mobile. On aura donc pas mal d'évolutions d'interface sur la prochaine release courant d'été

**Comment by camillemonchicourt** [comment]
Les intitulés ont été homogénéisés dans la 2.2.0-RC0, mais en effet "Ajouter un dénombrement" est trop long et ne rentre pas sur tous les appareils : 

![Screenshot_20220517-105845_Occtax](https://user-images.githubusercontent.com/4418840/168773467-50423035-8245-4579-bcac-2128baf4e47d.jpg)

**Comment by sgrimault** [comment]
Je vous propose donc la version "light" :
* "+ relevé"
* "+ dénombrement"
* "+ taxon"
sur l'ensemble des boutons d'ajout.

Dites moi si cela vous convient ?

**Comment by DonovanMaillard** [comment]
parfait pour +  et l'objet ajouté. Merci !

**Comment by DonovanMaillard** [comment]
Les boutons + ont été mis en place dans la version 2.2. 

Le bouton "Sauvegarder" a été placé en bas à droite depuis la version 2.3.0

L'affichage du formulaire de dénombrement sur une modale pleine page est conservé à ce stade et semble être la meilleure solution. 
La proposition d'avoir le formulaire sur la même page que la liste des dénombrements saisis pose divers soucis d'affichage et de complexités en termes de comportement des utilisateurs (risque de ne pas terminer une saisie, passer en mode édition sur un dénombrement alors que la saisie d'un autre n'est pas terminée, passer en mode édition alors qu'on veut simplement voir un dénombrement terminé etc).
